homeassistant/components/config/device_registry.py

```diff
@@ -138,10 +138,14 @@ async def websocket_remove_config_entry_from_device(
             "Failed to remove device entry, rejected by integration"
         )
 
-    entry = registry.async_update_device(
-        device_id, remove_config_entry_id=config_entry_id
-    )
+    # Integration might have removed the config entry already, that is fine.
+    try:
```

Comment:
The `KeyError` feels like an implementation detail leaking out. Can you use `if registry.async_get(device_id):`

Reply:
This one was discussed in the Core meeting, and most in there actually wanted the opposite. As a matter of fact, your suggestion was the original implementation.